ut_core.py

```diff
@@ -0,0 +1,149 @@
+
+import socket
+import time
+import yaml
+import re
+
+class UTCoreMenuNavigator:
+    """
+    Navigates through the UTcore menu system, trigger the execution of test cases, and collect results.
+    """
+
+    def __init__(self, menu_config_path, console, test_profile_url=None):
+        """
+        Initializes the UTCoreMenuNavigator object with a menu configuration file and an optional test profile.
+
+        Args:
+            menu_config_path (str): The file path to the menu configuration YAML file.
+            console: The console session object to communicate with the UTcore system.
+            test_profile_url (str, optional): The URL to a test profile YAML file. Defaults to None.
+        """
+        self.session = console
+        self.menu_config = self.load_yaml(menu_config_path)
+        self.shell = self.session.open_interactive_shell()
+
+        if test_profile_url:
+            self.test_profile = self.load_yaml(test_profile_url)
+        else:
+            self.test_profile = None
```

Comment:
I think this will happen outside of this class. So we've already got the menu_config and test binaries ready and we pass those into this class.